File: app/src/main/java/bobzien/com/smartrecyclerviewadapter/ViewHolder/NumViewHolder.java
package bobzien.com.smartrecyclerviewadapter.ViewHolder;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import bobzien.com.smartrecyclerviewadapter.R;
import bobzien.com.smartrecyclerviewadapter.SmartRecyclerViewAdapter.ViewHolder;
import butterknife.ButterKnife;
import butterknife.InjectView;

public class NumViewHolder extends ViewHolder<Integer>
{
    @InjectView(R.id.text)
    protected TextView textView;

    public NumViewHolder(@NonNull Context context, @NonNull Class clazz, @Nullable View itemView)
    {
        super(context, clazz, itemView);

        if(itemView != null)
        {
            ButterKnife.inject(this, itemView);
        }
    }

    @Override
    public ViewHolder getInstance(@NonNull View itemView)
    {
        return new NumViewHolder(getContext(), getClass(), itemView);
    }

    @Override
    public int getLayoutResourceId()
    {
        return R.layout.list_item_num;
    }

    @NonNull
    @Override
    public void bindViewHolder(@NonNull Integer item, boolean selected)
    {
        textView.setText(item + "");
    }
}

File: app/src/main/kotlin/bobzien/com/smartrecyclerviewadapter/ViewHolder/NumViewHolder.kt
package bobzien.com.smartrecyclerviewadapter.ViewHolder

import android.content.Context
import android.view.View
import android.widget.TextView
import bobzien.com.smartrecyclerviewadapter.R
import com.marmor.smartrecyclerviewadapter.SmartRecyclerViewAdapter.ViewHolder

class NumViewHolder(context: Context, clazz: Class<Integer>, itemView: View?) : ViewHolder<Integer>(context, clazz, itemView) {
    protected lateinit var textView: TextView

    init {
        if (itemView != null) {
            textView = itemView.findViewById(R.id.text) as TextView
        }
    }

    override fun getInstance(itemView: View): ViewHolder<Any> {
        return NumViewHolder(context, getHandledClass() as Class<Integer>, itemView) as ViewHolder<Any>
    }

    override val layoutResourceId: Int
        get() = R.layout.list_item_num

    override fun bindViewHolder(item: Integer, selected: Boolean) {
        textView.text = item.toString()
    }
}
